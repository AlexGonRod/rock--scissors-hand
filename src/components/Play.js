import React, {useState} from "react";
import Header from './Header';
import Nav from './Nav';
import Game from './Game';
import Icon from './Icon';
const choices = ['rock', 'paper', 'scissors'];

const Play = ({ score, setScore, user }) => {
  const [myChoice, setMyChoice] = useState('')
  const [computerChoice, setComputerChoice] = useState('')

  const [counter, setCounter] = useState(2)

  const newHousePick = () => {
		setComputerChoice(choices[Math.floor(Math.random() * 3)])
	}
  
const handlerSetChoice = (e) => {
    setCounter(1)
    setMyChoice(e.target.dataset.id);
    newHousePick();
}

  return (
		<>
		  <Nav user={user }/>
			<Header score={score} />
			<div className="play">
				<div className="items">
					{choices.map((item, index) => (
						<Icon
							key={index}
							dataId={`${item}`}
							onClick={handlerSetChoice}
							className={`icon icon-play-${item}`}
						/>
					))}
				</div>
				{myChoice && (
					<Game
						myChoice={myChoice}
						score={score}
						setScore={setScore}
						computerChoice={computerChoice}
						counter={counter}
						setCounter={setCounter}
					/>
				)}
			</div>
		</>
	)
}

export default Play;
