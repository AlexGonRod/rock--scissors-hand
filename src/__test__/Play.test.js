import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'
import Play from '../components/Play';

describe('<Play />', () => {
    let wrapper = shallow(<Play />);
    beforeEach(() => {
        wrapper = shallow(<Play />);
    });
    test('<Play /> renders', () => {
        expect(wrapper).toMatchSnapshot();
    })
});