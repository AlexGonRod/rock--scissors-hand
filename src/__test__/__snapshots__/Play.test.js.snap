// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Play /> <Play /> renders 1`] = `
<Fragment>
  <Nav />
  <Header />
  <div
    className="play"
  >
    <div
      className="items"
    >
      <Icon
        className="icon icon-play-rock"
        dataId="rock"
        key="0"
        onClick={[Function]}
      />
      <Icon
        className="icon icon-play-paper"
        dataId="paper"
        key="1"
        onClick={[Function]}
      />
      <Icon
        className="icon icon-play-scissors"
        dataId="scissors"
        key="2"
        onClick={[Function]}
      />
    </div>
  </div>
</Fragment>
`;
